libs/vm-modules/src/math/random.cpp

```diff
@@ -68,15 +68,15 @@ fetch::math::meta::IfIsNotFixedPoint128<T, T> Rand(VM *vm, T const &a = T{.0}, T
   if (a >= b)
   {
     vm->RuntimeError("Invalid argument: rand(a, b) must satisfy a < b");
-    return T{.0};
+    return fetch::math::AsType<T>(0.0);
   }
 
   std::random_device rd;
   std::mt19937_64    mt(rd());
 
   auto a_dbl = static_cast<double>(a);
   auto b_dbl = static_cast<double>(b);
-  return static_cast<T>(std::uniform_real_distribution<double>{a_dbl, b_dbl}(mt));
+  return fetch::math::AsType<T>(std::uniform_real_distribution<double>{a_dbl, b_dbl}(mt));
```

Comment:
This is under IFIsNotFixedPoint SFINAE so we KNOW that T(double) will work just fine. So we don't need AsType, right?

Reply:
this is actually IfIsNotFixedPoint128, so it's a FixedPoint, but not a FixedPoint128, which is handled below, a bit confusing name for a SFINAE I agree.